### PR TITLE
[KYUUBI-1819] Support closing Flink SQL engine process

### PR DESCRIPTION
**Pull request by SteNicholas**
### _Why are the changes needed?_

Support closing Flink SQL engine process.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by SteNicholas** [comment]
@yaooqinn @pan3793 @yanghua, could you please help to review?

**Comment by SteNicholas** [comment]
@yaooqinn @pan3793 @yanghua[](https://github.com/SteNicholas), could you please help to review?

File: kyuubi-server/src/test/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilderSuite.scala [hunk 1]
```diff
@@ -28,4 +28,24 @@ class FlinkProcessBuilderSuite extends KyuubiFunSuite {
...
     assert(commands.exists(_ endsWith "flink-sql-engine.sh"))
   }
+
+  test("kill application") {
+    val processBuilder = new FakeFlinkProcessBuilder(conf) {
+      override protected def env: Map[String, String] = Map("FLINK_HOME" -> "")
+    }
+    val exit1 = processBuilder.killApplication("""
+      [INFO] SQL update statement has been successfully submitted to the cluster:
+      Job ID: 6b1af540c0c0bb3fcfcad50ac037c862
```
Comment (pan3793) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilderSuite.scala:38]: what's the magic code?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala [hunk 2]
```diff
@@ -99,6 +102,29 @@ class FlinkProcessBuilder(
...
+            pb.redirectOutput(Redirect.appendTo(engineLog))
+            val process = pb.start()
+            process.waitFor() match {
+              case id if id != 0 => s"Failed to kill Application $jobId, please kill it manually. "
+              case _ => s"Killed Application $jobId successfully. "
+            }
+          case None =>
+            s"FLINK_HOME is not set! Failed to kill Application $jobId, please kill it manually."
+        }
+      case None => ""
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala:124]: log.warn the line

**Comment by pan3793** [comment]
We need an integration test case in `kyuubi-flink-it` module

File: kyuubi-server/src/test/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilderSuite.scala [hunk 3]
```diff
@@ -28,4 +28,24 @@ class FlinkProcessBuilderSuite extends KyuubiFunSuite {
...
+      """.stripMargin)
```
Comment (pan3793) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilderSuite.scala:39]:
```
"""
|
|""".stripMargin
```

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
> @pan3793, does the Spark engine also need the integration test for killing application? cc @yaooqinn

+1、the kill process now is hardcoded with yarn and its applicationid，which also need to be fixed

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Comment by ulysses-you** [comment]
@SteNicholas please create a new pr for `Spark engine`, thank you !

**Pull request closed without merging**